src/integration_tests/integration_tests.py

```diff
@@ -25,28 +25,52 @@ def test_dataset(client, bucket_loader):
     )
     assert dataset_metadata_response.status_code == 200
 
+    mock_unit_response = {
+        "busdesc": "Provision of equipment for hobbit adventures",
+        "local_unit": [
+            {
+                "luaddr2": "Underhill",
+                "luref": "2012763A",
+                "busdesc": "Creates old fashioned looking paper maps",
+                "luname": "Maps Factory",
+                "luaddr1": "1 Bag End",
+                "tradstyle": "Also Does Adventures Ltd",
+                "luaddr3": "Hobbiton",
+                "lupostcode": "HO1 1AA",
+            },
+            {
+                "luaddr2": "Maggotsville",
+                "luref": "20127364B",
+                "busdesc": "Quality pipe manufacturer",
+                "buslref": "pipe123",
+                "luname": "Pipes R Us Subsidiary",
+                "luaddr1": "12 The Farmstead",
+                "luaddr3": "Hobbiton",
+                "lupostcode": "HO1 1AB",
+            },
+        ],
+        "payeref": "123AB456",
+        "runame": "Pipes and Maps Ltd",
+        "rupostcode": "HO1 1AA",
+        "ruaddr1": "111 Under Hill",
+        "ruaddr4": "The Shire",
+        "ruaddr2": "Hobbitton",
+        "ruref": "43532",
+    }
     # Since the cloud function generates the GUID which is set as the dataset id, the below looping is necessary to
     # locate the specific dataset in the collection.
     # Iterate over all the items in the above API response, then locate the document with the "filename" field from above.
-    for guid, integration_dataset in dataset_metadata_response.json()[
-        "supplementary_dataset"
-    ].items():
-        if (
-            dataset_metadata_response.json()["supplementary_dataset"][guid]["filename"]
-            == filename
-        ):
-            dataset_id = guid
+    for dataset_metadata in dataset_metadata_response.json():
+        if dataset_metadata["filename"] == filename:
+            dataset_id = dataset_metadata["dataset_id"]
             unit_id = "43532"
             response = client.get(
                 f"/v1/unit_data?dataset_id={dataset_id}&unit_id={unit_id}"
             )
             assert response.status_code == 200
-            # Check that the API response is the same as the dataset just located in the loop
-            assert response.json() == integration_dataset
+            # Check that the API response is the same as the mock unit response
```

Comment:
You can probably delete this comment, the code is clear enough in what it does and it means you don't have to change the comment and the assertion every time there's a code change